src/main.rs

```diff
@@ -88,17 +88,25 @@ fn print_page(
     for path in page.paths() {
         let file = File::open(path).map_err(|msg| format!("Could not open file: {}", msg))?;
         let reader = BufReader::new(file);
+        let stdout = std::io::stdout();
+        let mut handle = stdout.lock();
 
         if enable_markdown {
             // Print the raw markdown of the file.
             for line in reader.lines() {
-                println!("{}", line.unwrap());
+                writeln!(handle, "{}", line.unwrap())
+                    .map_err(|_| "Could not write to stdout".to_string())?;
             }
         } else {
             // Create tokenizer and print output
             let mut tokenizer = Tokenizer::new(reader);
-            print_lines(&mut tokenizer, &config);
+            print_lines(&mut handle, &mut tokenizer, &config)
+                .map_err(|_| "Could not write to stdout".to_string())?;
```

Comment:
I think that this could use the specific error:

```
print_lines(&mut handle, &mut tokenizer, &config)
                .map_err(|e| format!("Could not write to stdout: {}", e)?;
```

This would mean that there should be an `impl` of the `Display` trait for `TealdeerError`, but that would just use the `.msg()` method you just wrote. (you could also use `e.msg()` here)